<a href="https://colab.research.google.com/github/Mado-k/introcomp/blob/main/Aula07.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

![](https://drive.google.com/uc?id=1hKhlJd2wJtTH5nPhuin48WbHeKNchO_b)

#  **Introdução**

No notebook anterior, você aprendeu a controlar o fluxo dos seus programas com a estrutura de repetição (`while`), permitindo que o código executasse um bloco de instruções continuamente enquanto uma condição específica fosse verdadeira. Agora, chegou a hora de juntar tudo o que você viu até aqui!

Ao longo do curso, você está construindo conhecimento. É como se cada conceito fosse um tijolo, e a base da construção precisa estar firme para sustentar obras mais complexas. Essa revisão vai servir para consolidar o seu conhecimento através da prática, revisitando os *tipos de dados*, a *aritmética*, a *entrada e saída* de dados e as *estruturas condicionais*, além da última ferramenta: o laço de repetição `while`.

Lembre-se: **errar faz parte do processo de aprendizado**. Não tenha medo de tentar e conte com a equipe do Introcomp para solucionar as dúvidas que aparecerem!

# **Como Interpretar um Problema**

Antes de iniciarmos a revisão, é necessário que você entenda a linha de raciocínio por trás da interpretação de um problema.

Para isso, é bom ter em mente esse pequeno passo a passo para te guiar na resolução dos exercícios.

## **Passo 1: Identificar as Entradas e Saídas**
* **Entrada:** O que o programa precisa receber? Quantas variáveis? Quais os tipos dados (`int`, `float`, `str`)?
* **Saída:** O que o programa precisa imprimir? Qual o texto exato do `print()`?

## **Passo 2: Desenhar a Lógica em Português**
Escreva a solução como uma receita antes de programar:
> *"Vou ler a nota 1 e a nota 2. Vou calcular a média. Se a média for maior ou igual a 6.0, imprimo 'Aprovado'. Senão, imprimo 'Reprovado'."*

## **Passo 3: Traduzir para Código Python**
Agora, você transforma esse algoritmo em código utilizando a sintaxe que aprendeu.

## **Exemplo Resolvido Passo a Passo:**
Crie um programa que receba o nome do aluno e sua nota em uma prova. Se a nota for maior ou igual a 7.0, exiba: `<Nome>, você passou!`. Caso contrário, exiba: `<Nome>, você ficou de recuperação!`.

**Aplicando os passos:**
1. **Entradas:** 2 entradas $\rightarrow$ `nome` (`str`) e `nota` (`float`).
2. **Saída:** Frase formatada com o nome.
3. **Lógica:** `if nota >= 7.0`.

In [ ]:
# Passo 3 na prática:
nome = input()
nota = float(input())

if nota >= 7.0:
    print(f"{nome}, você passou!")
else:
    print(f"{nome}, você ficou de recuperação!")

Caso você tenha dificuldade em assuntos iniciais do curso, leia o notebook de [Revisão Extra](https://colab.research.google.com/drive/1Ejwt6FhH_MXOz4c3aRp-QPvnVGdVnEtA?authuser=5#scrollTo=PUN8sHym4eeL) que fizemos justamente para esclarecer suas dúvidas.

# **Tipos de Dados e Aritmética**

## **Tipos de Dados**

Você já conhece os quatro tipos primitivos fundamentais do Python. Relembre o que cada um representa:

**`int` (inteiro):** Números inteiros, positivos ou negativos, sem casas decimais. Exemplos: idade de uma pessoa, quantidade de itens, número de páginas de um livro.

**`float` (ponto flutuante):** Números com casas decimais, usados para medidas precisas. Exemplos: altura, peso, temperatura, preço de produtos.

**`str` (string):** Sequências de caracteres, representam texto. Sempre ficam entre aspas (simples ou duplas). Exemplos: nomes, endereços, mensagens.

**`bool` (booleano):** Valores lógicos que representam apenas duas possibilidades: `True` (verdadeiro) ou `False` (falso). Exemplos: aprovado/reprovado, ligado/desligado, presente/ausente.

Utilize os tipos de dados no exercício que segue.

### **Exercício 1**

Imagine que você precisa criar um sistema para armazenar informações básicas dos alunos do projeto Introcomp. Para cada aluno, você deve guardar as seguintes informações:

- **nome:** Como o aluno se chama
- **altura:** Medida da altura em metros (exemplo: 1.75)
- **idade:** Quantos anos o aluno tem
- **código de matrícula:** Identificação única (exemplo: UFES2025123)

Para cada uma dessas quatro variáveis, identifique qual tipo (`int`, `float`, `str` ou `bool`) seria mais adequado para armazenar as informações, pensando na natureza de cada dado.

In [ ]:
# Complete o texto abaixo com os tipos correspondentes:

#nome ->
#altura ->
#idade ->
#codigo_matricula ->

# Agora, faça os Typecasts necessários para fazer a leitura correta das informações:

nome = input("Digite seu nome: ")
altura = input("Digite sua altura: ")
idade = input("Digite sua idade: ")
codigo = input("Digite seu código: ")

## **Aritmética**

Assim como na escola, é muito comum usar a matemática para solucionar problemas. Para fazer isso no Python, basta lembrar das operações:

In [ ]:
# Soma
soma = 2 + 2

# Subtração
subtracao = 20 - 5

# Multiplicação
multiplicacao = 4 * 3

# Divisão
divisao = 50 / 2

# Resto
resto = 10 % 3

# Potência
potencia = 6**2

print(soma)
print(subtracao)
print(multiplicacao)
print(divisao)
print(resto)
print(potencia)

As regras aqui são as mesmas da matemática, com o uso dos parênteses alterando a hierarquia das operações (ou seja, podem modificar qual operação vai ser feita primeiro). Para exercitar, resolva o problema de aquecimento e então faça o exercício proposto.

### **Aquecimento**

Joãozinho é muito sortudo e ganhou o sorteio de uma caixa de doces. Como ele é generoso, resolveu repartir o prêmio entre ele e seus 4 amigos, com cada um ganhando uma quantidade igual de guloseimas. Agora, ele precisa da sua ajuda para saber quantos doces vão restar depois que ele terminar a partilha do prêmio! Seu programa deve esperar uma entrada com a quantidade de doces, e então imprimir quantos sobraram após a divisão. Observe os casos de teste:

| Entrada                | Saída                                                                                      |
| ---------------------- | ------------------------------------------------------------------------------------------ |
| 10  | 0 |
| 8  | 3 |
| 23  | 3 |

In [4]:
# Ajude o Joãozinho a dividir seus doces
doces = int(input())
resto = doces % 5

print(resto)

23
3


### **Exercício 2**

Você está planejando uma viagem de carro e precisa calcular quantos litros de combustível serão necessários e quanto gastará no total. Para isso, você tem as seguintes informações:

- Distância total da viagem (ida e volta)
- Consumo do seu carro (quantos quilômetros ele faz com 1 litro)
- Preço do litro de combustível no posto

Seu programa deve calcular quantos litros serão necessários (distância dividida pelo consumo) e o custo total da viagem (litros × preço por litro). Para tanto, considere o seguinte modelo de entrada e saída:

**Entrada**
- Distância total da viagem em quilômetros
- Consumo do carro em km/L
- Preço por litro do combustível

**Saída:**
```
=== CÁLCULO DE COMBUSTÍVEL ===
Litros necessários: [valor] L
Custo total: R$ [valor]
```

**Casos de Teste**


| Entrada                | Saída                                                                                      |
| ---------------------- | ------------------------------------------------------------------------------------------ |
| 600 <br> 12 <br> 5.50  | === CÁLCULO DE COMBUSTÍVEL === <br> Litros necessários: 50.00 L <br> Custo total: R\$ 275.00 |
| 800 <br> 10 <br> 6.00  | === CÁLCULO DE COMBUSTÍVEL === <br> Litros necessários: 80.00 L <br> Custo total: R\$ 480.00 |
| 450 <br> 15 <br> 5.20  | === CÁLCULO DE COMBUSTÍVEL === <br> Litros necessários: 30.00 L <br> Custo total: R\$ 156.00 |
| 1000 <br> 20 <br> 4.99 | === CÁLCULO DE COMBUSTÍVEL === <br> Litros necessários: 50.00 L <br> Custo total: R\$ 249.50 |


In [8]:
# Escreva o código aqui para planejar sua viagem!
dist = float(input())
consumo = float(input())
preco_litro = float(input())

litros_necessários = dist / consumo
custo_litro = litros_necessários * preco_litro

print('=== CÁLCULO DE COMBUSTÍVEL ===')
print(f'Litros necessários: {litros_necessários} L')
print(f'Custo total: R$ {custo_litro}')

600
12
5.50
=== CÁLCULO DE COMBUSTÍVEL ===
Litros necessários: 50.0 L
Custo total: R$ 275.0


# **Entrada e Saída**

Dominar entrada e saída de dados é fundamental para criar programas que realmente interajam com o usuário, que sejam capazes de receber informações ou mostrar resultados. Pense num sistema de pedidos de um restaurante: o garçom precisa inserir o pedido do cliente (`input`) e o sistema deve exibir o total a pagar (`print`).

Essa troca de informações é essencial para programas usados no mundo real! Para consolidar esse conteúdo, resolva o próximo exercício.

### **Exercício 3**



Você está ajudando um cozinheiro a criar um sistema digital para compartilhar suas receitas. O programa deve esperar que o chef informe três ingredientes principais e o tempo de preparo, e então imprimir a receita seguindo o modelo especificado. Observe o padrão de entrada e saída abaixo:

**Entrada esperada:**
- Primeiro ingrediente
- Segundo ingrediente  
- Terceiro ingrediente
- Tempo de preparo em minutos

**Saída Correspondente**
```
=== RECEITA DO CHEF ===
Ingredientes principais:
- [primeiro ingrediente]
- [segundo ingrediente]
- [terceiro ingrediente]
Tempo de preparo: [tempo] minutos
Bom apetite!
```

**Caso de Teste**

| Entrada | Saída |
| :--- | :--- |
| Trigo<br>Ovos<br>Leite<br>35 | === RECEITA DO CHEF ===<br>Ingredientes principais:<br>- Trigo<br>- Ovos<br>- Leite<br>Tempo de preparo: 35 minutos<br>Bom apetite! |

In [9]:
# Ajude o chef! Escreva a solução aqui
first_ing = str(input())
second_ing = str(input())
third_ing = str(input())
preparing_time = int(input())

print('=== RECEITA DO CHEF ===')
print('Ingredientes principais:')
print(f'- {first_ing}')
print(f'- {second_ing}')
print(f'- {third_ing}')
print(f'Tempo de preparo: {preparing_time}')

TRIGO
OVOS
LEITE
35
=== RECEITA DO CHEF ===
Ingredientes principais:
- TRIGO
- OVOS
- LEITE
Tempo de preparo: 35


# **Operadores Lógicos**

Comparações são parte fundamental da programação, pois é por meio delas que você irá modelar problemas do mundo real. Relembre o significado dos operadores:

```
A > B: A maior que B
A < B: A menor que B
A >= B: A maior ou igual a B
A <= B: A menor ou igual a B
A == B: A igual a B
A != B: A diferente de B
```

Além disso, relações lógicas são construídas com base em operadores lógicos:
- **and**, quando todos os valores testados devem ser verdadeiros
- **or**, quando basta que um valor seja verdadeiro
- **not**, quando se quer inverter o valor de uma variável ou expressão

Exercite esse conteúdo com a questão abaixo.

### **Exercício 4**


A jovem Bárbara gosta muito de ir a restaurantes, mas só consegue fazer isso sob algumas condições. Primeiro de tudo: o saldo em sua conta deve ser **superior** a R$ 200,00. Além disso, as seguintes regras devem ser levadas em conta:
- **Todos** de sua família (composta pelo pai, mãe e irmã) vão ao restaurante com ela, **ou**
- Sua amiga Aline vai ao restaurante com ela

Para treinar suas habilidades, faça um programa que diga se Bárbara pode ou não ir ao restaurante, de modo que ela informe seu saldo, a disponibilidade de sua família, a disponibilidade de sua amiga Aline, e então confira se pode sair.

**Entrada**:
- O saldo de Bárbara
- A disponibilidade de sua irmã
- A disponibilidade de sua mãe
- A disponibilidade do seu pai
- A disponibilidade de Aline

A disponibilidade das pessoas pode valer `1` (caso em que a pessoa **pode** ir ao restaurante) ou `0` (caso em que a pessoa **não pode** ir ao restaurante). Se a ida de Bárbara ao restaurante for possível, seu código deve imprimir `True`; do contrário, deve imprimir `False`. Observe o exemplo de entrada e saída correspondente:

| Entrada                                                                 | Saída  |
|-------------------------------------------------------------------------|--------|
| 250 <br> 0 <br> 0 <br> 0 <br> 0                                         | False  |
| 500 <br> 1 <br> 1 <br> 1 <br> 0                                         | True   |
| 200 <br> 1 <br> 1 <br> 1 <br> 1                                         | False  |
| 700 <br> 1 <br> 0 <br> 1 <br> 1                                         | True   |


In [20]:
saldo = float(input())
sis = int(input())
mom = int(input())
father = int(input())
aline = int(input())
pode_ir = False

if ((sis and mom and father) == 1 or aline == 1) and saldo > 200:
  pode_ir = True
print(pode_ir)

700
1
0
1
1
True


#**Condicionais**

Com os operadores condicionais `if`, `else` e `elif`, é possível resolver problemas mais complexos e realistas, criando códigos que se adaptam a diferentes situações. Com essas novas ferramentas, você deixa de criar programas que sempre fazem a mesma coisa, para criar códigos pensados para analisar cenários e se comportar de acordo com as circunstâncias, como jogos, sistemas de avaliação, entre outros. Relembre a modelagem com estruturas condicionais a partir do fluxograma abaixo:

<div align="center">
  <img src="https://drive.google.com/uc?id=1pw-1HBG5eG7vQDR5o4rj0EDwFGBuNguA" width="800"/>
</div>

<center>
<b>Figura 1:</b> Fluxograma de uma estrutura condicional.
</center>

Para reforçar esse tema, resolva os próximos exercícios.

### **Exercício 5**

Sua escola precisa de um programa para determinar se um aluno foi aprovado ou reprovado, sob uma regra simples: notas iguais ou superiores a 7.0 resultam em aprovação. Observe os exemplos de entrada e saída:

| Entrada | Saída |
| :--- | :--- |
| Amilton<br>6.7 | Amilton tirou 6.7 e vai fazer recuperação. |
| Roberta<br>9.5 | Roberta tirou 9.5 e passou direto. |

In [22]:
# Faça aqui o seu sistema de avaliação
aluno = input()
nota = float(input())

if nota >= 7.0:
  print(f'{aluno} tirou {nota} e passou direto.')
elif nota < 7.0:
  print(f'{aluno} tirou {nota} e vai fazer recuperação.')

Roberta
9.5
Roberta tirou 9.5 e passou direto.


### **Exercício 6**

A mesma escola decidiu refinar seu sistema de avaliação, criando mais categorias para o desempenho dos alunos, considerando notas com duas casas decimais:
- Nota inferior a 4.00: "Reprovado"
- Nota de 4.00 a 6.99: "Recuperação"
- Nota de 7.00 a 8.99: "Aprovado"
- Nota de 9.00 a 10.00: "Aprovado com Louvor"

A entrada e saída esperada seguirão o seguinte padrão:

**Entrada:**
- Nome do aluno
- Nota final

**Saída:**
- [Nome do aluno] - Situação: [situação]

**Casos de Teste**

| Entrada              | Saída                                      |
|----------------------|--------------------------------------------|
| Doroteia <br> 3.5 | Doroteia - Situação: Reprovado              |
| João Vitor <br> 6 | João Vitor - Situação: Recuperação          |
| Cláudia <br> 8.99 | Cláudia - Situação: Aprovado                |
| Jorge <br> 9   | Jorge - Situação: Aprovado com Louvor        |



In [27]:
# Melhore o sistema do exercício anterior
aluno = input()
nota = float(input())
situacao = 'Andamento'

if 7.0 <= nota < 9:
  situacao = 'Aprovado'
  print(f'{aluno} - Situação: {situacao}')
elif nota < 7.0:
  situacao = 'Reprovado'
  print(f'{aluno} - Situação: {situacao}')
elif nota >= 9:
  situacao = 'Aprovado com louvor'
  print(f'{aluno} - Situação: {situacao}')

Cláudia
8.99
Cláudia - Situação: Aprovado


### **Exercício 7**

Você já fez um código para calcular o custo com gasolina numa viagem, mas e se você precisasse que ele fosse mais robusto? Agora, ele deve analisar se você tem dinheiro suficiente para a viagem. Incremente o código feito no exercício 2 levando em conta o seguinte fluxo:

Inicialmente, seu código deve esperar as seguintes entradas:

- Distância total da viagem (em Km)
- Consumo do seu carro (quantos quilômetros ele faz com 1 litro)
- Preço do litro de combustível no posto
- Quanto dinheiro você pode gastar com gasolina

Seu programa deve calcular quantos litros serão necessários (distância dividida pelo consumo) e o custo total da viagem (litros × preço por litro). Após isso, seu código deve imprimir `Gastos com gasolina: [valor total da gasolina]` e então os seguintes caminhos são possíveis:

* O custo da gasolina ultrapassa seu saldo
  - Nesse caso, o código deve imprimir `A viagem não será possível`.
* Seu saldo é suficiente
  - Nesse cenário, seu código deve esperar mais uma entrada, representando a quantia de dinheiro reservada para outros gastos da viagem, e o valor que sobrou da gasolina deve ser acrescentado nesse orçamento. Após isso, seu código deve imprimir `Gasolina comprada, agora seu orçamento é de [valor]`.


**Casos de Teste**


| Entrada                | Saída                                                                                      |
| ---------------------- | ------------------------------------------------------------------------------------------ |
| 600 <br> 12 <br> 5.50 <br> 300 <br> 650 | Gastos com gasolina: 275<br>Gasolina comprada, agora seu orçamento é de 675 |
| 800 <br> 10 <br> 6.00 <br> 400 | Gastos com gasolina: 480<br>A viagem não será possível |

In [32]:
# Adapte aqui o código feito anteriormente
dist = float(input())
consumo = float(input())
preco_litro = float(input())
saldo = float(input())

litros_necessários = dist / consumo
custo_litro = litros_necessários * preco_litro
gasto = custo_litro

if gasto < saldo:
  orcamento = saldo - gasto
  print(f'Gastos com gasolina: {gasto}\nGasolina comprada, agora seu orçamento é de {orcamento}')
elif gasto > saldo:
  print(f'Gastos com gasolina: {gasto}\nA viagem não será possível')

800
10
6.00
400
Gastos com gasolina: 480.0
A viagem não será possível


# **While**

O `while` é uma estrutura de repetição que permite que o seu código processe informações **enquanto** uma determinada condição for verdadeira. Dessa forma, o código fica mais enxuto, dinâmico e consegue lidar com uma variedade maior de situações.

``` python
while <condição é verdadeira>:
    <comando_1>
    <comando_2>
    ...
```

<div align="center">
  <img src="https://drive.google.com/uc?id=11p5CW-6omVWSh--ToUd-bUZGTWaILYCX" width="400"/>
</div>


<center>
<b>Figura 2:</b> Fluxograma da estrutura while.
</center>

### **Exercício 8**

Imagine que você está programando a lógica principal de pontuação para um pequeno jogo 2D. O seu personagem começa a fase com 0 moedas.

Crie um programa que pergunte repetidamente ao jogador: "Você pegou uma moeda? (s/n): ".

Enquanto a resposta for **"s"**, o programa deve adicionar **10 pontos** à pontuação total do jogador e exibir a mensagem: **"Mais 10 pontos! Pontuação atual: X"**.

Se o jogador digitar **"n"**, ou qualquer outra coisa, o ciclo deve ser quebrado.

Ao final, quando o loop terminar, o programa deve exibir: **"Fim de jogo! Sua pontuação final foi de X pontos."**.

**Caso de teste**:
| Entrada | Saída Esperada |
| :--- | :--- |
| s<br>s<br>s<br>n | Mais 10 pontos! Pontuação atual: 10<br>Mais 10 pontos! Pontuação atual: 20<br>Mais 10 pontos! Pontuação atual: 30<br>Fim de jogo! Sua pontuação final foi de 30 pontos. |

In [37]:
resp = str(input('Você pegou uma moeda? (s/n):'))
pts = 0
while resp.lower() == 's':
  pts += 10
  resp = str(input('Você pegou uma moeda? (s/n):'))
if pts < 100:
  print(f'Fim de jogo! Sua pontuação final foi de {pts} pontos.')
else:
  print(f'Fim de jogo! Sua pontuação final foi de {pts} pontos.\n E você é competitivo! Parabéns!')

Você pegou uma moeda? (s/n):s
Você pegou uma moeda? (s/n):s
Você pegou uma moeda? (s/n):s
Você pegou uma moeda? (s/n):s
Você pegou uma moeda? (s/n):s
Você pegou uma moeda? (s/n):s
Você pegou uma moeda? (s/n):s
Você pegou uma moeda? (s/n):s
Você pegou uma moeda? (s/n):s
Você pegou uma moeda? (s/n):s
Você pegou uma moeda? (s/n):s
Você pegou uma moeda? (s/n):n
Fim de jogo! Sua pontuação final foi de 110 pontos.
 E você é competitivo! Parabéns!


### **Exercício 9**

Desenvolva um programa que solicite ao usuário a entrada de um número inteiro positivo. O objetivo do seu código é calcular e exibir na tela todos os divisores desse número, exceto por 1 e pelo próprio número, em ordem crescente. Lembrete: Um número $X$ é considerado divisor de um número $Y$ se a divisão de $Y$ por $X$ for exata, ou seja, se o resto da divisão for igual a zero.

**Casos de teste:**
| Entrada | Saída Esperada |
| :--- | :--- |
| 12 | 2<br>3<br>4<br>6 |
| 28 | 2<br>4<br>7<br>14 |
| 7 | *(Nenhuma saída, pois 7 é primo e os únicos divisores são 1 e 7)* |

In [80]:
n = int(input())
i = 1

while True:
  if (n % i) == 0 and n != i != 1:
    print('--->',i)
  elif i == n1:
    break
  else:
    print(f'(Nenhuma saída, pois {n} é primo e os únicos divisores são 1 e {n})')
    break
  i += 1


113
(Nenhuma saída, pois 7 é primo e os únicos divisores são 1 e 113)


### **Exercício 10**

Crie um pequeno simulador de processamento de dados de um sensor de temperatura. O programa deve solicitar repetidamente que o usuário insira uma "leitura do sensor" (um número inteiro). O loop deve rodar continuamente enquanto o usuário não digitar um valor negativo. O valor negativo servirá apenas como um "botão de parada" e não deve ser contabilizado nos cálculos.

Além disso, caso o valor da temperatura lida seja maior que **60°C**, o programa deve iniciar um processo de resfriamento, diminuindo a temperatura até que atinja **30°C** ou menos. Para isso, o programa deve exibir na tela a temperatura diminuindo de **5°C** em **5°C**. Os valores do resfriamento não devem ser contados no cálculo final.

Quando o usuário digitar um número negativo e o loop for encerrado, o programa deve imprimir a quantidade de leituras válidas realizadas e a temperatura média. Caso a primeira coisa que o usuário digite seja um número negativo, o programa deve exibir o aviso: "Nenhum dado válido foi inserido para calcular a média."

**Casos de teste**:
| Entrada        | Saída Esperada                                         |
| :---           | :---                                                   |
| -1             | --- Relatório Final ---<br>Nenhum dado válido foi inserido para calcular a média. |
| 40<br>50<br>-5 | --- Relatório Final ---<br>Total de leituras válidas: 2<br>Temperatura média: 45.00°C|
| 65<br>-1       | Iniciando processo de resfriamento...<br>60°C<br>55°C<br>50°C<br>45°C<br>40°C<br>35°C<br>30°C<br><br>--- Relatório Final ---<br>Total de leituras válidas: 2<br>Temperatura média: 47.50°C |

In [126]:
# Desenvolva aqui o sensor!
leitura = 0
l1 = leitura
leitura_total = 0
i = 0

while True:
  leitura = float(input())
  if leitura < 0 and i == 0:
    print('--- Relatório Final ---\n Nenhum dado válido foi inserido para calcular a média')
    break
  elif leitura > 0:
    leitura_total += leitura
    i += 1
  elif leitura < 0 and i > 1:
    temp_media = leitura_total / i
    print(f'--- Relatório Final ---\nTotal de leituras válidas: {i}\nTemperatura média: {temp_media:.2f}°C')
    break

  elif leitura > 60:
    print('Iniciando processo de resfriamento...')
    while leitura > 30:
      leitura -= 5
      print(f'{leitura}°C')

40
50
-1
--- Relatório Final ---
Total de leituras válidas: 2
Temperatura média: 45.00°C


#**Conclusão**

# **O que Você Revisou**

Você revisou hoje:
- Como os **tipos primitivos** servem para representar informações de diferentes naturezas;
- Aplicou **operações aritméticas** para resolver problemas matemáticos práticos;
- Estabeleceu a comunicação entre programa e usuário por meio de **entrada e saída** de dados;
- Explorou como as **estruturas condicionais** tornam seus programas mais inteligentes;
- Relembrou o laço de repetição **while**, que permite repetir uma série de comandos enquanto uma condição se mantiver verdadeira, deixando seus códigos mais enxutos e dinâmicos.

Tais conceitos servem de base para qualquer programa mais complexo e o domínio desses fundamentos é essencial para sua evolução como programador, pois eles aparecem constantemente em projetos maiores e mais desafiadores.

Continue praticando e experimentando, conte com o suporte da equipe Introcomp e boa prova semana que vem!

# **Resolução dos Exercícios**

### **Exercício 1**

In [ ]:
# Complete o texto abaixo com os tipos correspondentes:

#nome -> str
#altura -> float
#idade -> int
#codigo_matricula -> str

### **Aquecimento**

In [ ]:
num_pessoas = 5    # Dito  no enunciado
qtd_doces = int(input("Digite a quantidade de doces: "))

resto = qtd_doces % num_pessoas

print(resto)

Digite a quantidade de doces: 600
0


### **Exercício 2**

In [ ]:
# Entradas
distancia = float(input("Digite a distância: "))
consumo = float(input("Digite o consumo: "))
preco = float(input("Digite o preço: "))

# Cálculos necessários
litros_gastos = distancia/consumo
valor_gasto = litros_gastos * preco

# Saída formatada
print('=== CÁLCULO DE COMBUSTÍVEL ===')
print(f'Litros necessários: {litros_gastos:.2f} L')
print(f'Custo total: R$ {valor_gasto:.2f}')

Digite a distância: 600
Digite o consumo: 12
Digite o preço: 5.5
=== CÁLCULO DE COMBUSTÍVEL ===
Litros necessários: 50.00 L
Custo total: R$ 275.00


### **Exercício 3**

In [ ]:
ingrediente1 = input("Digite o 1º ingrediente: ")
ingrediente2 = input("Digite o 2º ingrediente: ")
ingrediente3 = input("Digite o 3º ingrediente: ")
tempo = int(input("Digite o tempo: "))
print('=== RECEITA DO CHEF ===')
print('Ingredientes principais:')
print(f'- {ingrediente1}')
print(f'- {ingrediente2}')
print(f'- {ingrediente3}')
print(f'Tempo de preparo: {tempo} minutos')
print('Bom apetite!')

Digite o 1º ingrediente: ovo
Digite o 2º ingrediente: fermento
Digite o 3º ingrediente: farinha
Digite o tempo: 10
=== RECEITA DO CHEF ===
Ingredientes principais:
- ovo
- fermento
- farinha
Tempo de preparo: 10 minutos
Bom apetite!


###**Exercício 4**

In [ ]:
# Curiosidade: o número 0 equivale a 'False' e
# qualquer numero diferente dele vale 'True'

# Entradas
saldo = float(input("Saldo disponível: "))
irma = bool(int(input("Irmã vai? ")))
mae = bool(int(input("Mãe vai? ")))
pai = bool(int(input("Pai vai? ")))
aline = bool(int(input("Aline vai? ")))

# Regras ditas no enunciado
pode_ir = (saldo > 200) and ((irma and mae and pai) or aline)

print(pode_ir)


Saldo disponível: 300
Irmã vai? 0
Mãe vai? 0
Pai vai? 0
Aline vai? 0
False


### **Exercício 5**

In [ ]:
nome = input("Digite o nome: ")
nota = float(input("Digite a nota: "))

if nota >= 7.0:
    print(f"{nome} tirou {nota:.2f} e passou direto.")
else:
    print(f"{nome} tirou {nota:.2f} e vai fazer recuperação.")

Digite o nome: ana
Digite a nota: 6.8
ana tirou 6.80 e vai fazer recuperação.


### **Exercício 6**

In [ ]:
# Essa não é a única forma possível de resolver essa questão!

nome = input("Digite o nome: ")
nota = float(input("Digite a nota: "))

if nota < 4.00:
    situacao = "Reprovado"
elif nota < 7.00:
    situacao = "Recuperação"
elif nota < 9.00:
    situacao = "Aprovado"
else:
    situacao = "Aprovado com Louvor"

print(f"{nome} - Situação: {situacao}")

Digite o nome: ana
Digite a nota: 6.8
ana - Situação: Recuperação


### **Exercício 7**

In [ ]:
# Entradas
distancia = float(input("Digite a distância: "))
consumo = float(input("Digite o consumo: "))
preco = float(input("Digite o preço: "))
orcamento_gasolina = float(input("Digite o orçamento: "))

# Cálculos necessários
litros_gastos = distancia/consumo
valor_gasto = litros_gastos * preco

print(f'Gastos com gasolina: {valor_gasto}')

if valor_gasto > orcamento_gasolina:
  print('A viagem não será possível')
else:
  # Note uma entrada que só ocorre sob essa condicão
  orcamento_geral = float(input("Digite o orçamento geral: "))
  sobra = orcamento_gasolina - valor_gasto
  orcamento_geral = orcamento_geral + sobra

  print(f'Gasolina comprada, agora seu orçamento é de {orcamento_geral:.2f}')


Digite a distância: 600
Digite o consumo: 12
Digite o preço: 5.5
Digite o orçamento: 300
Gastos com gasolina: 275.0
650
Gasolina comprada, agora seu orçamento é de 675.00


### **Exercício 8**

In [ ]:
pontuacao = 0
continuar = "s"

print("Bem-vindo à fase de coleta de moedas!")

# O loop roda enquanto a variável 'continuar' for exatamente igual a "s"
while continuar == "s":
    continuar = input("Você pegou uma moeda? (s/n): ")

    if continuar == "s":
        pontuacao += 10
        print(f"Mais 10 pontos! Pontuação atual: {pontuacao}")

print(f"\nFim de jogo! Sua pontuação final foi de {pontuacao} pontos.")

Bem-vindo à fase de coleta de moedas!
Você pegou uma moeda? (s/n): s
Mais 10 pontos! Pontuação atual: 10
Você pegou uma moeda? (s/n): s
Mais 10 pontos! Pontuação atual: 20
Você pegou uma moeda? (s/n): s
Mais 10 pontos! Pontuação atual: 30
Você pegou uma moeda? (s/n): e

Fim de jogo! Sua pontuação final foi de 30 pontos.


### **Exercício 9**

In [ ]:
numero = int(input("Digite um número inteiro positivo: "))
print(f"Os divisores de {numero} são: ")


# O loop vai testar todos os números de 2 até o próprio (número digitado - 1)
i = 2
while(i < numero):
    # Se o resto da divisão for zero, 'i' é um divisor válido
    if numero % i == 0:
        print(i)
    i+=1

Digite um número inteiro positivo: 15
Os divisores de 15 são: 
3
5


###**Exercício 10**

In [ ]:
soma_temperaturas = 0
quantidade_leituras = 0

print("Insira as leituras do sensor. Digite um valor negativo para encerrar.")

while True:
    leitura = int(input("Leitura do sensor: "))

    # Condição de parada imediata
    if leitura < 0:
        break

    # Contabiliza o valor da temperatura atual
    soma_temperaturas += leitura
    quantidade_leituras += 1

    # Processo de resfriamento
    if leitura > 60:
        print("\nIniciando processo de resfriamento...")
        while leitura > 30:
            leitura -= 5
            print(f"{leitura}°C")

print("\n--- Relatório Final ---")
if quantidade_leituras > 0:
    media = soma_temperaturas / quantidade_leituras
    print(f"Total de leituras válidas: {quantidade_leituras}")
    print(f"Temperatura média: {media:.2f}°C")
else:
    print("Nenhum dado válido foi inserido para calcular a média.")

Insira as leituras do sensor. Digite um valor negativo para encerrar.
Leitura do sensor: 65

Iniciando processo de resfriamento...
60°C
55°C
50°C
45°C
40°C
35°C
30°C
Leitura do sensor: -1

--- Relatório Final ---
Total de leituras válidas: 1
Temperatura média: 65.00°C
